Mathlib/Algebra/Polynomial/RingDivision.lean

```diff
@@ -337,33 +338,42 @@ end Ring
 
 section CommSemiring
 
-variable [CommSemiring R]
+variable [CommSemiring R] {a p : R[X]}
 
-theorem Monic.C_dvd_iff_isUnit {p : R[X]} (hp : Monic p) {a : R} :
-    C a ∣ p ↔ IsUnit a :=
+section Monic
+
+variable (hp : p.Monic)
+
+theorem Monic.C_dvd_iff_isUnit {a : R} : C a ∣ p ↔ IsUnit a :=
   ⟨fun h => isUnit_iff_dvd_one.mpr <|
       hp.coeff_natDegree ▸ (C_dvd_iff_dvd_coeff _ _).mp h p.natDegree,
    fun ha => (ha.map C).dvd⟩
 
-theorem degree_pos_of_not_isUnit_of_dvd_monic {a p : R[X]} (ha : ¬ IsUnit a)
-    (hap : a ∣ p) (hp : Monic p) :
-    0 < degree a :=
-  lt_of_not_ge <| fun h => ha <| by
-    rw [Polynomial.eq_C_of_degree_le_zero h] at hap ⊢
-    simpa [hp.C_dvd_iff_isUnit, isUnit_C] using hap
-
-theorem natDegree_pos_of_not_isUnit_of_dvd_monic {a p : R[X]} (ha : ¬ IsUnit a)
-    (hap : a ∣ p) (hp : Monic p) :
-    0 < natDegree a :=
-  natDegree_pos_iff_degree_pos.mpr <| degree_pos_of_not_isUnit_of_dvd_monic ha hap hp
-
-theorem degree_pos_of_monic_of_not_isUnit {a : R[X]} (hu : ¬ IsUnit a) (ha : Monic a) :
-    0 < degree a :=
-  degree_pos_of_not_isUnit_of_dvd_monic hu dvd_rfl ha
-
-theorem natDegree_pos_of_monic_of_not_isUnit {a : R[X]} (hu : ¬ IsUnit a) (ha : Monic a) :
-    0 < natDegree a :=
-  natDegree_pos_iff_degree_pos.mpr <| degree_pos_of_monic_of_not_isUnit hu ha
+theorem Monic.natDegree_pos_of_ne_one (hu : p ≠ 1) : 0 < natDegree p := by
+  by_contra! hp'
+  exact hu <| eq_one_of_monic_natDegree_zero hp (by omega)
```

Comment:
I think you can golf this by first making `eq_one_of_monic_natDegree_zero` an iff (which you could rename to `Monic.natDegree_eq_zero`, which you can itself prove by replacing the use of `eq_C_of_natDegree_eq_zero` by `eq_C_coeff_zero_iff_natDegree_eq_zero` in the existing proof.